TestPathHelper signature change causing issue on JBeret ecosystem CI

The innocuous-looking change in https://github.com/quarkusio/quarkus/pull/46625 is causing an issue running the JBatch TCK: https://github.com/quarkiverse/quarkus-jberet/actions/runs/13757259497

With:

```
java.lang.RuntimeException: java.lang.IllegalStateException: Unable to translate path for /home/runner/.m2/repository/jakarta/batch/com.ibm.jbatch.tck/2.1.1/com.ibm.jbatch.tck-2.1.1.jar
	at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:611)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:706)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.IllegalStateException: Unable to translate path for /home/runner/.m2/repository/jakarta/batch/com.ibm.jbatch.tck/2.1.1/com.ibm.jbatch.tck-2.1.1.jar
	at io.quarkus.test.common.PathTestHelper.getAppClassLocationForTestLocation(PathTestHelper.java:225)
	at io.quarkus.test.junit.AbstractJvmQuarkusTestExtension.createAugmentor(AbstractJvmQuarkusTestExtension.java:137)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:190)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:578)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:628)
	... 1 more
```

--- a/test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java
+++ b/test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java
@@ -178,9 +178,10 @@ public final class PathTestHelper {
      * @return directory or JAR containing the application being tested by a test from the given location
      */
     public static Path getAppClassLocationForTestLocation(Path testClassLocationPath) {
-        if (testClassLocationPath.endsWith(".jar")) {
-            if (testClassLocationPath.endsWith("-tests.jar")) {
-                String testClassLocation = testClassLocationPath.toString();
+        String testClassLocation = testClassLocationPath.toString();
+
+        if (testClassLocation.endsWith(".jar")) {
+            if (testClassLocation.endsWith("-tests.jar")) {
                 return Paths.get(new StringBuilder()
                         .append(testClassLocation, 0, testClassLocation.length() - "-tests.jar".length())
                         .append(".jar")
@@ -188,7 +189,6 @@ public final class PathTestHelper {
             }
             return testClassLocationPath;
         }
-        String testClassLocation = testClassLocationPath.toString();
         Optional<Path> mainClassesDir = TEST_TO_MAIN_DIR_FRAGMENTS.entrySet().stream()
                 .filter(e -> testClassLocation.contains(e.getKey()))
                 .map(e -> {
@@ -211,7 +211,7 @@ public final class PathTestHelper {
         }
         // could be a custom test classes dir under the 'target' dir with the main
         // classes still under 'target/classes'
-        p = Path.of(testClassLocation).getParent();
+        p = testClassLocationPath.getParent();
         if (p != null && p.getFileName().toString().equals(TARGET)) {
             p = p.resolve("classes");
             if (Files.exists(p)) {

